src/flb_engine.c

```diff
@@ -723,8 +755,8 @@ int flb_engine_start(struct flb_config *config)
     }
 
     while (1) {
-        mk_event_wait(evl);
-        mk_event_foreach(event, evl) {
+        mk_event_wait(evl); /* potentially conditional mk_event_wait or mk_event_wait_2 based on bucket queue capacity for one shot events */
```

Comment:
If we expect people to register edge triggered events into the event loop in the future, then this may be a problem.

Let's say that in the last loop round, the bucket queue collected an edge triggered event, and the loop exited due to max iteration (which is meant to allow the cleanup functions to run periodically even in high throughput cases)
Then mk_event_wait(evl) will wait until the next event is picked up since edge triggered events will only be detected once.

This can be easily resolved with the following. I didn't want to change the logic in the engine too much though, and it seems that only level triggered events are used so far which means that this will not be a problem, so I decided against this.
```
if (bucket_queue_is_empty) {
  mk_event_wait(evl);
}
else {
  mk_event_wait_2(evl, 0);
}
```